DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DSACampaignsServlet.java

```diff
@@ -61,16 +67,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        UserService userService = UserServiceFactory.getUserService();
+        boolean hangoutsRequest = request.getParameter("hangouts") != null
+        UserService userService = !hangoutsRequest ? UserServiceFactory.getUserService() : null;
 
         if (request.getParameter("delete") != null) {
             System.err.println("Inside If Parameter");
             String campaignId = request.getParameter("id");
             deleteDSACampaign(campaignId);
             return;
-        } else if (userService.isUserLoggedIn()) {
+        } else if (hangoutsRequest || userService.isUserLoggedIn()) {
```

Comment:
Can you separate this code into multiple else if statements, it's getting too complicated with statements like UserService userService = !hangoutsRequest ? UserServiceFactory.getUserService() : null; 
Something like
else if (hangoutsRequest) {
} else if (userService.isUserLoggedIn()) {
}
This would let you remove a lot of your if statements.

Reply:
Added

Reply:
Fixed